tad/services/storage.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC, abstractmethod
+from pathlib import Path
+
+from yaml import dump
+
+
+class WriterFactory(ABC):
+    @staticmethod
+    def get_writer(writer_type: str = "file", **kwargs):
+        match writer_type:
+            case "file":
+                return FileSystemWriteService(location=kwargs["location"], filename=kwargs["filename"])
```

Comment:
You might want to check in kwargs contains location and throw an nice error if not in kwargs